gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -624,7 +624,8 @@ public SeekableByteChannel open(
         resourceId.isStorageObject(), "Expected full StorageObject id, got %s", resourceId);
 
     if (storageOptions.isGrpcEnabled()) {
-      return GoogleCloudStorageGrpcReadChannel.open(storageStubProvider, resourceId, readOptions);
+      return GoogleCloudStorageGrpcReadChannel
```

Comment:
I think it will be a good idea to have a separate constructor for GrpcReadChannel, one that exists as of today which internally reads metadata via gRPC, and other which we will introduce in this PR (passing the additional storage and errorExtractor) which will read metaData via JSON API.

Later when getMetaData via gRPC is supported, we only will need to switch the constructor/initializer.

Reply:
Since the constructors are public, we would not want to expose an implementation which isn't functional

Reply:
It does bring up a good point of backward compatibility, Since we do not support gRPC as GA, I would assume, it is ok to break the constructor.

@davidrabinowitz  Please advice

Reply:
As the grpc only implementation is broken at this time, let's remove it and force the new constructor